instrumentation/net/http/otelhttp/config.go

```diff
@@ -141,6 +143,14 @@ func WithSpanOptions(opts ...trace.SpanStartOption) Option {
 	})
 }
 
+// WithMetricAttributes adds a function that will be called to add attributes
+// to the request metrics.
```

Comment:
This should include a warning to avoid high cardinality metrics unless the user is sure that their destination system can handle them.